### DISCUSSION TITLE
Mutual Blocking in C++ Multi-threaded Inference with CompiledModel Instances

### DISCUSSION THREAD

**Discussion by EunhaTaker** [openvinotoolkit/openvino#24760] [Q&A]
When c++ multiple threads use a CompiledModel singleton, or when using multiple instances to infer, they both encounter mutual blocking issues. Can this be avoided through configuration? Would improper model compilation lead to this situation?

**Comment by ilya-lavrenov** [comment]
Do you use sync `request.infer()` call on multiple requests of the same `ov;:CompiledModel`?
If yes, prefer `request.start_async`

**Reply by EunhaTaker** [reply]
Thank you, I'll give it a try when I have some time.
